lesson5-LINQ/Task1/LinqTask.cs

```diff
@@ -8,39 +9,38 @@ public static class LinqTask
     {
         public static IEnumerable<Customer> Linq1(IEnumerable<Customer> customers, decimal limit)
         {
-            throw new NotImplementedException();
+            var result = customers.Where(c => c.Orders.Sum(o => o.Total) > limit).ToList();
+            return result;
         }
 
-        public static IEnumerable<(Customer customer, IEnumerable<Supplier> suppliers)> Linq2(
-            IEnumerable<Customer> customers,
-            IEnumerable<Supplier> suppliers
-        )
+        public static IEnumerable<(Customer customer, IEnumerable<Supplier> suppliers)> Linq2(IEnumerable<Customer> customers, IEnumerable<Supplier> suppliers)
         {
-            throw new NotImplementedException();
+            var result = customers.Select(c => (c, suppliers.Where(s => s.Country == c.Country && s.City == c.City)));
+            return result;
         }
 
-        public static IEnumerable<(Customer customer, IEnumerable<Supplier> suppliers)> Linq2UsingGroup(
-            IEnumerable<Customer> customers,
-            IEnumerable<Supplier> suppliers
-        )
+        public static IEnumerable<(Customer customer, IEnumerable<Supplier> suppliers)> Linq2UsingGroup(IEnumerable<Customer> customers, IEnumerable<Supplier> suppliers)
         {
-            throw new NotImplementedException();
+            var result = customers.GroupJoin(suppliers, customer => new { customer.City, customer.Country },
+                 innerSupplier => new { innerSupplier.City, innerSupplier.Country },
+                 (selectedCustomer, innerSupplier) => (selectedCustomer, innerSupplier));
+            return result;
         }
 
         public static IEnumerable<Customer> Linq3(IEnumerable<Customer> customers, decimal limit)
         {
-            throw new NotImplementedException();
+            var result = customers.Where(c => c.Orders.Any(o => o.Total > limit));
+            return result;
         }
 
-        public static IEnumerable<(Customer customer, DateTime dateOfEntry)> Linq4(
-            IEnumerable<Customer> customers
-        )
+
+        public static IEnumerable<(Customer customer, DateTime dateOfEntry)> Linq4(IEnumerable<Customer> customers)
         {
-            throw new NotImplementedException();
+            var result = customers.Where(c => c.Orders.Length != 0).Select(c => (c, c.Orders.Min(o => o.OrderDate)));
+            return result;
         }
 
-        public static IEnumerable<(Customer customer, DateTime dateOfEntry)> Linq5(
-            IEnumerable<Customer> customers
+        public static IEnumerable<(Customer customer, DateTime dateOfEntry)> Linq5(IEnumerable<Customer> customers
         )
         {
             throw new NotImplementedException();
```

Comment:
Please put the tasks text itself in the comments  so I will be aware of what you exactly fixing.

Reply:
done